c/secp256k1_blake160_sighash_all.c

```diff
@@ -94,70 +115,89 @@ int main(int argc, char* argv[]) {
     witness_pos.ptr = (const uint8_t*)witness;
     witness_pos.size = len;
 
-    /* Load signature */
-    arg_res = mol_cut(&witness_pos, MOL_Witness(0));
-    if (arg_res.code != 0) {
-      if (arg_res.attr < 1) {
-        return ERROR_WRONG_NUMBER_OF_ARGUMENTS;
-      } else {
-        return ERROR_ENCODING;
-      }
-    }
-    witness_len = arg_res.attr;
-
-    bytes_res = mol_cut_bytes(&arg_res.pos);
-    if (bytes_res.code != 0) {
-      return ERROR_ENCODING;
-    } else if (bytes_res.pos.size < 65) {
-      return ERROR_ENCODING;
-    }
+    /* verify threshold signatures */
+    memset(used_signatures, 0, sigs_cnt);
 
-    /* The 65th byte is recid according to contract spec.*/
-    recid = bytes_res.pos.ptr[RECID_INDEX];
-    /* Recover pubkey */
-    secp256k1_ecdsa_recoverable_signature signature;
-    if (secp256k1_ecdsa_recoverable_signature_parse_compact(
-            &context, &signature, bytes_res.pos.ptr, recid) == 0) {
-      return ERROR_SECP_PARSE_SIGNATURE;
-    }
-    blake2b_state blake2b_ctx;
-    blake2b_init(&blake2b_ctx, BLAKE2B_BLOCK_SIZE);
-    blake2b_update(&blake2b_ctx, tx_hash, BLAKE2B_BLOCK_SIZE);
-    for (size_t i = 1; i < witness_len; i++) {
+    for (size_t i = 0; i < threshold; i++) {
+      /* Load signature */
       arg_res = mol_cut(&witness_pos, MOL_Witness(i));
       if (arg_res.code != 0) {
-        return ERROR_ENCODING;
+        if (arg_res.attr < threshold) {
+          return ERROR_WRONG_NUMBER_OF_ARGUMENTS;
+        } else {
+          return ERROR_ENCODING;
+        }
       }
+      witness_len = arg_res.attr;
```

Comment:
If `witness_len < threshold`, we can return early here.

Reply:
we already check the length in https://github.com/nervosnetwork/ckb-system-scripts/pull/35/files/dcaffdd0c8e5633e3504ee44352acde53600a5ec#diff-ccd9a3a9b2781124b082417ea9746eb4R125

Reply:
In 125, you only check the length when the mol failed to get a slice of witness Vec.

Reply:
https://github.com/nervosnetwork/ckb-system-scripts/pull/35#discussion_r316133203

Reply:
I see, the `if` expression is wrong.